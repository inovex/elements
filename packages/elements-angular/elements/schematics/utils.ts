import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
