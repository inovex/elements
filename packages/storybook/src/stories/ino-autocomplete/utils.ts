import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
