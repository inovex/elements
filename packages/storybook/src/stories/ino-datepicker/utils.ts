import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
