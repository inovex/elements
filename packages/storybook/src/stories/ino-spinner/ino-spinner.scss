.flex-parent {
  display: flex;
  flex-wrap: wrap;

  > .flex-child {
    margin: 0 20px;
  }
}