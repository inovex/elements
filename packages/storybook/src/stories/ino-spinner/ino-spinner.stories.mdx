import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-spinner/readme.md";

<Meta title="Notification/ino-spinner/Documentation" />

<Description markdown={readme} />
