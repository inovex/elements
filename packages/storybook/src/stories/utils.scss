/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
