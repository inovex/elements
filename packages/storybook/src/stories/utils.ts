import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
