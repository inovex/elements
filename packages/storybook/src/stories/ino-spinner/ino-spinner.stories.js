import { storiesOf } from '@storybook/html';

import { number, boolean, select } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-spinner/readme.md';
import './ino-spinner.scss';



storiesOf('Notification|<ino-spinner>', module)
  .addDecorator(withStencilReadme(componentReadme))
  .add('Default usage', () => /*html*/`
    <div class="story-spinner">
      <div class="flex-parent-center">
        <div>
          <h4>Customizable Spinner</h4>
          <ino-spinner
            class="customizable-spinner"
            ino-color-scheme="${select(
              'ino-color-scheme',
              ['primary', 'secondary', 'success', 'warning', 'error', 'light', 'dark' ],
              'primary'
            )}"
            ino-type="${select(
              'ino-type',
              ['tile', 'bounce', 'circle'],
              'tile'
            )}"
            ino-height="${number('ino-height', 40)}"
            ino-width="${number('ino-width', 40)}"
            ino-modal="${boolean('ino-modal', false)}"
          >
          </ino-spinner>
        </div>
      </div>

      <h4>Color Scheme</h4>

      <div class="flex-parent">
        <div class="flex-child">
          <h5>primary</h5>
          <ino-spinner ino-color-scheme="primary"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>secondary</h5>
          <ino-spinner ino-color-scheme="secondary"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>success</h5>
          <ino-spinner ino-color-scheme="success"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>warning</h5>
          <ino-spinner ino-color-scheme="warning"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>error</h5>
          <ino-spinner ino-color-scheme="error"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>light</h5>
          <ino-spinner ino-color-scheme="light"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>dark</h5>
          <ino-spinner ino-color-scheme="dark"></ino-spinner>
        </div>
      </div>

      <h4>Type</h4>
      <div class="flex-parent">
        <div class="flex-child">
          <h5>tile</h5>
          <ino-spinner ino-type="tile"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>bounce</h5>
          <ino-spinner ino-type="bounce"></ino-spinner>
        </div>
        <div class="flex-child">
          <h5>circle</h5>
          <ino-spinner ino-type="circle"></ino-spinner>
        </div>
      </div>
    </div>
  `)
  .add('As modal', () => /*html*/`
    <div class="story-button">
      <ino-spinner ino-modal></ino-spinner>
    </div>
  `);
