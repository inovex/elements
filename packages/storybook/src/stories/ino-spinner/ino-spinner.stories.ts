import { Components } from '@inovex.de/elements';
import { html } from 'lit-html';
import './ino-spinner.scss';
import { defaultDecorator, withColorScheme } from '../utils';
import { Story } from '@storybook/web-components';

export default {
  title: 'Notification/ino-spinner',
  component: 'ino-spinner',
  decorators: [defaultDecorator],
};

export const Playground: Story<Components.InoSpinner> = (args) => html`
  <ino-spinner
    color-scheme="${args.colorScheme}"
    height="${args.height}"
    modal="${args.modal}"
    type="${args.type}"
    width="${args.width}"
  >
  </ino-spinner>
`;
Playground.args = {
  modal: false,
  type: 'tile',
  height: 40,
  width: 40,
};
Playground.argTypes = {
  type: {
    control: {
      type: 'select',
    },
    options: ['circle', 'tile', 'bounce'],
  },
};
withColorScheme(Playground, 'colorScheme', 'primary');

export const Colors = () => html`
  <div class="flex-parent">
    <div class="flex-child">
      <h5>primary</h5>
      <ino-spinner color-scheme="primary"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>secondary</h5>
      <ino-spinner color-scheme="secondary"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>success</h5>
      <ino-spinner color-scheme="success"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>warning</h5>
      <ino-spinner color-scheme="warning"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>error</h5>
      <ino-spinner color-scheme="error"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>light</h5>
      <ino-spinner color-scheme="light"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>dark</h5>
      <ino-spinner color-scheme="dark"></ino-spinner>
    </div>
  </div>
`;

export const Types = () => html`
  <div class="flex-parent">
    <div class="flex-child">
      <h5>tile</h5>
      <ino-spinner type="tile"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>bounce</h5>
      <ino-spinner type="bounce"></ino-spinner>
    </div>
    <div class="flex-child">
      <h5>circle</h5>
      <ino-spinner type="circle"></ino-spinner>
    </div>
  </div>
`;

// TODO: Breaks page
// export const AsModal = () => html`
//   <ino-spinner modal></ino-spinner>
// `;
