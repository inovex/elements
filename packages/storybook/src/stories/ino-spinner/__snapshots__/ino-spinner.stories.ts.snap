// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification/ino-spinner ColorScheme smoke-test 1`] = `
<div id="root-inner">
  <div class="story-spinner">
    <ino-spinner color-scheme="secondary"
                 height="40"
                 modal="false"
                 type="circle"
                 width="40"
                 class="ino-spinner--type-circle ino-spinner--color-scheme-secondary hydrated"
    >
      <div class="ino-spinner__composer"
           style="width: 40px; height: 40px;"
      >
        <div class="ino-spinner__circle-1">
        </div>
        <div class="ino-spinner__circle-2">
        </div>
        <div class="ino-spinner__circle-3">
        </div>
        <div class="ino-spinner__circle-4">
        </div>
        <div class="ino-spinner__circle-5">
        </div>
        <div class="ino-spinner__circle-6">
        </div>
        <div class="ino-spinner__circle-7">
        </div>
        <div class="ino-spinner__circle-8">
        </div>
        <div class="ino-spinner__circle-9">
        </div>
        <div class="ino-spinner__circle-10">
        </div>
        <div class="ino-spinner__circle-11">
        </div>
        <div class="ino-spinner__circle-12">
        </div>
      </div>
    </ino-spinner>
  </div>
</div>
`;

exports[`Notification/ino-spinner Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-spinner">
    <ino-spinner color-scheme="primary"
                 height="40"
                 modal="false"
                 type="tile"
                 width="40"
                 class="ino-spinner--type-tile ino-spinner--color-scheme-primary hydrated"
    >
      <div class="ino-spinner__composer"
           style="width: 40px; height: 40px;"
      >
        <div class="ino-spinner__rect1">
        </div>
        <div class="ino-spinner__rect2">
        </div>
        <div class="ino-spinner__rect3">
        </div>
        <div class="ino-spinner__rect4">
        </div>
        <div class="ino-spinner__rect5">
        </div>
      </div>
    </ino-spinner>
  </div>
</div>
`;

exports[`Notification/ino-spinner Type smoke-test 1`] = `
<div id="root-inner">
  <div class="story-spinner">
    <ino-spinner color-scheme="primary"
                 height="40"
                 modal="false"
                 type="bounce"
                 width="40"
                 class="ino-spinner--type-bounce ino-spinner--color-scheme-primary hydrated"
    >
      <div class="ino-spinner__composer"
           style="width: 40px; height: 40px;"
      >
        <div class="ino-spinner__double-bounce-1">
        </div>
        <div class="ino-spinner__double-bounce-2">
        </div>
      </div>
    </ino-spinner>
  </div>
</div>
`;
